server/app/mutations/grid_services/update.rb

```diff
@@ -75,28 +75,34 @@ def execute
       attributes[:health_check] = self.health_check if self.health_check
       attributes[:volumes_from] = self.volumes_from if self.volumes_from
 
+      embeds_dirrty = false
+
       if self.links
         attributes[:grid_service_links] = build_grid_service_links(
           self.grid_service.grid_service_links.to_a,
           self.grid_service.grid, grid_service.stack, self.links
         )
+        embeds_dirrty = attributes[:grid_service_links] != self.grid_service.grid_service_links.to_a
       end
 
       if self.hooks
         attributes[:hooks] = self.build_grid_service_hooks(self.grid_service.hooks.to_a)
+        embeds_dirrty = attributes[:hooks] != self.grid_service.hooks.to_a
       end
 
       if self.secrets
         attributes[:secrets] = self.build_grid_service_secrets(self.grid_service.secrets.to_a)
+        embeds_dirrty = attributes[:secrets] != self.grid_service.secrets.to_a
       end
       if self.volumes
         attributes[:service_volumes] = self.build_service_volumes(self.grid_service.service_volumes.to_a,
           self.grid_service.grid, self.grid_service.stack
         )
+        embeds_dirrty = attributes[:service_volumes] != self.grid_service.service_volumes.to_a
       end
       grid_service.attributes = attributes
 
-      if grid_service.changed?
+      if grid_service.changed? || embeds_dirrty
         info "updating service #{grid_service.to_path} with changes: #{changed(grid_service)}"
         grid_service.revision += 1
```

Comment:
Yes, this should work, because the `revision` change should be enough to actually mark the model as `changed?`, so that the `updated_at` also gets changed.